### ISSUE TITLE
Empty map returned for empty tag

### ISSUE THREAD

**Issue by gunnar2k** [homanchou/elixir-xml-to-map#9]
Hey, awesome library 😄 

But, what about, instead of:

```elixir
"<xml><lol/></xml>" 
|> XmlToMap.naive_map()
```

…outputting:

```elixir
%{"xml" => %{"lol" => %{}}}
```

it could output a `nil` instead of that empty map?

```elixir
%{"xml" => %{"lol" => nil}}
```

**Comment by homanchou** [comment]
Thanks. In regards to your question: i think it’s possible. A PR with a
test is welcome. 😉

Could you explain what use case this solves for you?

**Comment by gunnar2k** [comment]
Happy to submit a PR! 🙌 

The use case, for example, is when XML-based APIs (in my case the [Intacct API](https://developer.intacct.com/api/accounts-payable/bills/#create-bill)) might return an empty tag to indicate null-value:

```xml
...
<CURRENCY></CURRENCY>
...
```

…and when this gets returned as `%{currency: %{}}` , that empty map has to be converted to `nil` on the caller-side, so it can be safely passed to `Ecto`, for instance 🤷‍♂️  Would be nice if it was just returned as `nil` in the first place 😬 

Another, similar use-case, also for the [Intacct API for error-handling](https://developer.intacct.com/web-services/error-handling/) when the API might return

```xml
...
<errorno>Some #</errorno>
<description>Some description</description>
<description2></description2>
<correction></correction>
...
```

… and interpolating those things into 

```elixir
"#{errorno} – #{description}, #{description2}"
``` 

throws error if one of the optional values are a map:

```elixir
 (Protocol.UndefinedError) protocol String.Chars not implemented for %{} of type Map
```

So like I said, I'll be happy to submit a PR for this if you'd be willing to merge it :pray:

**Comment by gunnar2k** [comment]
Created a PR now :pray: https://github.com/homanchou/elixir-xml-to-map/pull/10

**Comment by homanchou** [comment]
Thanks for the explanation.

That makes sense.  At first I was worried that returning nil would be less
consistent api compared to always returning a map. It’s like saying I have
empty map of children, which recursively pans out.  However, if a leaf node
contains a value we don’t return a map either, we return the value.

And also making this more compatible with ecto seems like a plus.

Can you also bump the version major in your PR?  As this would  be a
breaking change in behavior.  I’ve lost internet at home for a week (I’m
limited to using my phone for emails for now. Service guy is allegedly
coming Monday).  I’ll merge whenever i get my internet back so i can deploy
the updated package to hex.

Thanks!

**Comment by gunnar2k** [comment]
Awesome! Bumped the major in the PR now to 2.0.0 🎉 

Thanks!

**Comment by homanchou** [comment]
Merged and 2.0.0 hex package published.  Thanks!  ❤️ 💜 💟  💙 ❤️
